python/delta_sharing/reader.py

```diff
@@ -105,30 +122,70 @@ def _copy(
 
     @staticmethod
     def _to_pandas(
-        add_file: AddFile, converters: Dict[str, Callable[[str], Any]], limit: Optional[int]
+        action: FileAction,
+        converters: Dict[str, Callable[[str], Any]],
+        for_cdc: bool,
+        limit: Optional[int],
     ) -> pd.DataFrame:
-        url = urlparse(add_file.url)
+        url = urlparse(action.url)
         if "storage.googleapis.com" in (url.netloc.lower()):
             # Apply the yarl patch for GCS pre-signed urls
             import delta_sharing._yarl_patch  # noqa: F401
 
         protocol = url.scheme
         filesystem = fsspec.filesystem(protocol)
 
-        pa_dataset = dataset(source=add_file.url, format="parquet", filesystem=filesystem)
+        pa_dataset = dataset(source=action.url, format="parquet", filesystem=filesystem)
         pa_table = pa_dataset.head(limit) if limit is not None else pa_dataset.to_table()
         pdf = pa_table.to_pandas(
             date_as_object=True, use_threads=False, split_blocks=True, self_destruct=True
         )
 
         for col, converter in converters.items():
             if col not in pdf.columns:
-                if col in add_file.partition_values:
+                if col in action.partition_values:
                     if converter is not None:
-                        pdf[col] = converter(add_file.partition_values[col])
+                        pdf[col] = converter(action.partition_values[col])
                     else:
                         raise ValueError("Cannot partition on binary or complex columns")
                 else:
                     pdf[col] = None
 
+        if for_cdc:
+            # Add the change type col name to non cdc actions.
+            if type(action) != AddCdcFile:
```

Comment:
nit: we can use `if for_cdf and type(action) != AddCdcFile`?

Reply:
I'm skipping processing timestamp/version processing as well.